### PR TITLE
Revert "Temporarily disable HIP build on Jenkins CI"

### PR DESCRIPTION
**Pull request by dalg24**
Revert #3205 
The machine is back online.

### CODE REVIEW

**Comment by dalg24** [comment]
The [log](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/2325/pipeline/37#step-91-log-17) does not print the architecture.

**Comment by dalg24** [comment]
> The [log](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/2325/pipeline/37#step-91-log-17) does not print the architecture.

[Fixed](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/2327/pipeline/37#step-88-log-17) in 1d0aa80

**Comment by dalg24** [comment]
> Looks good. I assume that you set `HIP_VISIBLE_DEVICES` on the tester appropriately.

Yes

File: .jenkins [hunk 1]
```diff
@@ -24,36 +24,36 @@ pipeline {
...
+                            args '-v /tmp/ccache.kokkos:/tmp/ccache --device=/dev/kfd --device=/dev/dri --security-opt seccomp=unconfined --group-add video --env HIP_VISIBLE_DEVICES=$HIP_VISIBLE_DEVICES'
+                        }
+                    }
+                    steps {
+                        sh 'ccache --zero-stats'
+                        sh '''rm -rf build && mkdir -p build && cd build && \
+                              cmake \
+                                -DCMAKE_BUILD_TYPE=Debug \
+                                -DCMAKE_CXX_COMPILER=hipcc \
+                                -DCMAKE_CXX_FLAGS="-Werror -Wno-unused-command-line-argument -Wno-braced-scalar-init" \
```
Comment (masterleinad) [inline .jenkins:43]: I think we can drop at least the last flag here now but I can try that in another pull request.

Comment (masterleinad) [inline .jenkins:43]:
```suggestion
                                -DCMAKE_CXX_FLAGS="-Werror -Wno-unused-command-line-argument" \
```

**Pull request merged**